src/type_checker.cr

```diff
@@ -222,6 +222,8 @@ module Mint
               return NEVER.as(Checkable)
             elsif node.is_a?(Ast::Function)
               static_type_signature(node)
+            elsif node.is_a?(Ast::WhereStatement)
+              static_type_signature_for_inline_function(node)
```

Comment:
We can keep `static_type_signature` method since crystal has overloading https://crystal-lang.org/reference/syntax_and_semantics/overloading.html

Reply:
I named it like that because it's not a type signature for WhereStatement but for InlineFunction IF it appears to be inside WhereStatement.

Reply:
Then maybe we should move checking the condition here? 

```
elsif node.is_a?(Ast::WhereStatement) && node.expression.is_a?(Ast::InlineFunction)
  static_type_signature(node)
````

I'm not sure if the compiler is smart enough to handle this.

Reply:
Nope, it is not.